nrfx/mdk/compiler_abstraction.h

```diff
@@ -150,7 +150,8 @@ POSSIBILITY OF SUCH DAMAGE.
 
     static inline unsigned int gcc_current_sp(void)
     {
-        register unsigned sp __ASM("sp");
+        register unsigned sp;
+        __ASM volatile("mov  %0, sp" : "=r" (sp) : : "memory");
```

Comment:
Is the "memory" clobber actually needed? The assembly code does not perform any read or write to a memory location, so why force the compiler to flush or reread anything?

Reply:
I think I modeled this on the CMSIS version.  @ioannisg thoughts on the 'memory' clobber?

Reply:
CMSIS implements the function with memory clobber as well:
```

/** \brief  Get Stack Pointer
    \return Stack Pointer value
 */
__STATIC_FORCEINLINE uint32_t __get_SP(void)
{
  uint32_t result;
  __ASM volatile("MOV  %0, sp" : "=r" (result) : : "memory");
  return result;
}
```

@anangl @galak I wonder if we can just use the CMSIS instead of own implementation

Reply:
@ioannisg But this snippet is from `Core_R/Include/cmsis_gcc.h`, right? For nRF chips we'd rather need to use `Core/Include/cmsis_gcc.h`.
Maybe @simtind could help here, as this will anyway need to be changed in MDK ultimately.

Reply:
@anangl true, I looked at the wrong file :)
Looking at cortex-m:
```
/**
  \brief   Get Process Stack Pointer
  \details Returns the current value of the Process Stack Pointer (PSP).
  \return               PSP Register value
 */
__STATIC_FORCEINLINE uint32_t __get_PSP(void)
{
  uint32_t result;

  __ASM volatile ("MRS %0, psp"  : "=r" (result) );
  return(result);
}
```

memory clobber seems not needed, probably OK to remove.

Reply:
Yeah, guess I was looking at the wrong version as well.

Now fixed - removed the memory clobber.